packages/rocketchat-lib/server/functions/saveUser.js

```diff
@@ -27,7 +34,7 @@ RocketChat.saveUser = function(userId, userData) {
 		});
 	}
 
-	if (userData.roles && _.indexOf(userData.roles, 'admin') >= 0 && !RocketChat.authz.hasPermission(userId, 'assign-admin-role')) {
+	if (userData.roles && _.indexOf(userData.roles, 'admin') >= 0 && !RocketChat.authz.hasRole('admin')) {
```

Comment:
Not validating the permission but the role dis-allows a non-admin with this permission to assign admins.
I agree that this permission is kind-of futile, but still I don't see a need to change this.